lib/pbench/server/database/models/datasets.py

```diff
@@ -366,6 +370,27 @@ def validate_access(self, key: str, value: str) -> str:
             return access
         raise DatasetBadParameterType(value, "access keyword")
 
+    @validates("owner")
+    def validate_owner(self, key: str, value: Any) -> User:
+        """
+        Validate and translate owner name to User object
+        Args:
+            key: owner
+            value: username
+        Raises:
+            DatasetBadParameter: the owner value given doesn't resolve to a
+                Pbench username.
+        Returns:
+            User object
+        """
+        if type(value) is User:
+            return value
+        elif type(value) is str:
+            user = User.query(username=value)
+            if user:
+                return user
+        raise DatasetBadParameterType(value, "username")
```

Comment:
The `owner` column is of User type (it's a sqlalchemy relationship actually) but we can specify `owner="username"` (as a part of convenience). We have this function with a decorator `@validates(owner)` this validation checks if the value is `user` object itself or a string if a string we expect it to be a `username` so we can return the correct `user` object by querying it with a username. The way `User.query` function works is that it takes either `username` or `id` and we have been using the username because it's the most readable among the two.

Reply:
Understood.

> The way `User.query` function works is that it takes either `username` or `id`

Right, so why doesn`t `validate_owner()` accept/support/try `id` as well?

Reply:
The actual DB column is `id`, which is now the OIDC user ID. We use "username" for incoming and outgoing references, and the `validate_owner` was a SQLAlchemy helper to do that translation. Sure, it could have been done elsewhere, but I don't see any point in changing it in this context. (Aside from the fact that originally `id` was `Integer` and `username` was `str` so it was easy to distinguish: now they're both `str` and we'd need to do a lookup, which I'm not even entirely sure SQLAlchemy allows in a validator, to distinguish.) In any case, worrying about this seems like unnecessary noise right now.

Reply:
> we'd need to do a lookup, which I'm not even entirely sure SQLAlchemy allows in a validator

Um...what are we doing here at line 389?  (All I'm proposing is another call, just like that one, if that one fails to produce a match.)

> In any case, worrying about this seems like unnecessary noise right now.

Fair enough.